File: app/src/main/java/ru/mammoth70/wherearetheynow/AppColors.java
package ru.mammoth70.wherearetheynow;

import java.util.ArrayList;
import java.util.Arrays;

public class AppColors {
    // Класс для работы с цветовыми ресурсами.

    public static final String COLOR_WHITE = "#FFFFFF";
    public static final String COLOR_BLACK = "#000000";
    public static final String COLOR_RED = "#FF001A";
    public static final String COLOR_ORANGE = "#FF5D00";
    public static final String COLOR_YELLOW = "#FFBB00";
    public static final String COLOR_GREEN = "#32FF00";
    public static final String COLOR_DARKGREEN = "#168000";
    public static final String COLOR_CYAN = "#00FFFB";
    public static final String COLOR_BLUE = "#002EFF";
    public static final String COLOR_VIOLET = "#8F00FF";
    public static final String COLOR_MAGENTA = "#FF00B1";

    public static final ArrayList<String> colors =
            new ArrayList<>(Arrays.asList(COLOR_RED, COLOR_ORANGE, COLOR_YELLOW, COLOR_GREEN, COLOR_DARKGREEN,
                    COLOR_CYAN, COLOR_BLUE, COLOR_VIOLET, COLOR_MAGENTA, COLOR_BLACK));

    private static final char OCTOHORPE = '#';
    private static final String COLOR_WHITE_8 = "#FFFFFFFF";
    private static final String ALPHA_48 = "48";
    private static final String ALPHA_16 = "16";

    public static String getColorAlpha(String color) {
        // Метод возвращает прозрачный цвет по строке с цветом.
        return addTransparenty(color, ALPHA_48);
    }

    public static String getColorAlpha16(String color) {
        // Метод возвращает прозрачный цвет по строке с цветом.
        return addTransparenty(color, ALPHA_16);
    }

    public static int getColorMarker(String color) {
        // Метод возвращает большую метку по строке с цветом.
        if (color != null) {
            switch (color) {
                case (COLOR_WHITE):
                    return R.drawable.ic_pin_white_64;
                case (COLOR_RED):
                    return R.drawable.ic_pin_red_64;
                case (COLOR_ORANGE):
                    return R.drawable.ic_pin_orange_64;
                case (COLOR_YELLOW):
                    return R.drawable.ic_pin_yellow_64;
                case (COLOR_GREEN):
                    return R.drawable.ic_pin_green_64;
                case (COLOR_DARKGREEN):
                    return R.drawable.ic_pin_darkgreen_64;
                case (COLOR_CYAN):
                    return R.drawable.ic_pin_cyan_64;
                case (COLOR_BLUE):
                    return R.drawable.ic_pin_blue_64;
                case (COLOR_VIOLET):
                    return R.drawable.ic_pin_violet_64;
                case (COLOR_MAGENTA):
                    return R.drawable.ic_pin_magenta_64;
                case (COLOR_BLACK):
                    return R.drawable.ic_pin_black_64;
                default:
                    return R.drawable.ic_pin_empty_64;
            }
        } return R.drawable.ic_pin_error_64;
    }

    public static int getColorMarkerSmall(String color) {
        // Метод возвращает маленькую метку по строке с цветом.
        if (color != null) {
            switch (color) {
                case (COLOR_WHITE):
                    return R.drawable.ic_pin_white_48;
                case (COLOR_RED):
                    return R.drawable.ic_pin_red_48;
                case (COLOR_ORANGE):
                    return R.drawable.ic_pin_orange_48;
                case (COLOR_YELLOW):
                    return R.drawable.ic_pin_yellow_48;
                case (COLOR_GREEN):
                    return R.drawable.ic_pin_green_48;
                case (COLOR_DARKGREEN):
                    return R.drawable.ic_pin_darkgreen_48;
                case (COLOR_CYAN):
                    return R.drawable.ic_pin_cyan_48;
                case (COLOR_BLUE):
                    return R.drawable.ic_pin_blue_48;
                case (COLOR_VIOLET):
                    return R.drawable.ic_pin_violet_48;
                case (COLOR_MAGENTA):
                    return R.drawable.ic_pin_magenta_48;
                case (COLOR_BLACK):
                    return R.drawable.ic_pin_black_48;
                default:
                    return R.drawable.ic_pin_empty_48;
            }
        } return R.drawable.ic_pin_error_48;
    }

    private static String addTransparenty(String color, String tranparent){
        // Функция добавляет прозрачность к цвету без прозрачности.
        // Если что-то не так - возвращает белый цвет.
        if ((color != null) && (tranparent != null) && (color.length() == 7) &&
                (tranparent.length() == 2) && (color.charAt(0) == OCTOHORPE)) {
            return OCTOHORPE + tranparent + color.substring(1, 7);
        } else {
            return COLOR_WHITE_8;
        }
    }

}

File: app/src/main/java/ru/mammoth70/wherearetheynow/AppColors.kt
package ru.mammoth70.wherearetheynow

import androidx.core.graphics.toColorInt

object AppColors {
    // Объект для работы с цветовыми ресурсами.

    const val COLOR_WHITE = "#FFFFFF"
    const val COLOR_BLACK = "#000000"
    const val COLOR_RED = "#FF001A"
    const val COLOR_ORANGE = "#FF5D00"
    const val COLOR_YELLOW = "#FFBB00"
    const val COLOR_GREEN = "#32FF00"
    const val COLOR_DARKGREEN = "#168000"
    const val COLOR_CYAN = "#00FFFB"
    const val COLOR_BLUE = "#002EFF"
    const val COLOR_LIGHTBLUE = "#4495FF"
    const val COLOR_VIOLET = "#8F00FF"
    const val COLOR_MAGENTA = "#FF00B1"
    const val COLOR_BROWN = "#7E4301"

    val colors =
        listOf(
            COLOR_RED, COLOR_ORANGE, COLOR_YELLOW, COLOR_GREEN, COLOR_DARKGREEN, COLOR_CYAN,
            COLOR_BLUE, COLOR_LIGHTBLUE, COLOR_VIOLET, COLOR_MAGENTA, COLOR_BROWN, COLOR_BLACK,
        )

    enum class Alpha(val tranparent: String) {
        ALPHA_48("#48"),
        ALPHA_16("#16")
    }

    fun getColorAlpha(color: String?): Int {
        // Функция возвращает номер прозрачного цвета по строке с цветом.
        return addTransparenty(color, Alpha.ALPHA_48).toColorInt()
    }

    fun getColorAlpha16(color: String?): Int {
        // Функция возвращает номер прозрачного цвета по строке с цветом.
        return addTransparenty(color, Alpha.ALPHA_16).toColorInt()
    }

    fun getMarker(color: String?): Int {
        // Функция возвращает маленькую метку по строке с цветом.
        color?.let {
            return when (color) {
                COLOR_WHITE -> R.drawable.ic_pin_white
                COLOR_RED -> R.drawable.ic_pin_red
                COLOR_ORANGE -> R.drawable.ic_pin_orange
                COLOR_YELLOW -> R.drawable.ic_pin_yellow
                COLOR_GREEN -> R.drawable.ic_pin_green
                COLOR_DARKGREEN -> R.drawable.ic_pin_darkgreen
                COLOR_CYAN -> R.drawable.ic_pin_cyan
                COLOR_BLUE -> R.drawable.ic_pin_blue
                COLOR_LIGHTBLUE -> R.drawable.ic_pin_lightblue
                COLOR_VIOLET -> R.drawable.ic_pin_violet
                COLOR_MAGENTA -> R.drawable.ic_pin_magenta
                COLOR_BROWN -> R.drawable.ic_pin_brown
                COLOR_BLACK -> R.drawable.ic_pin_black
                else -> R.drawable.ic_pin_empty
            }
        }
        return R.drawable.ic_pin_error
    }

    private fun addTransparenty(color: String?, alpha: Alpha): String {
        // Функция добавляет прозрачность к цвету без прозрачности.
        // Если что-то не так - возвращает белый цвет.
        return if ((!color.isNullOrBlank()) && (color.length == 7) && (color[0] == '#')) {
            alpha.tranparent + color.substring(1, 7)
        } else {
            "#FFFFFFFF"
        }
    }

}